docs/dev/fluent.rst

```diff
@@ -0,0 +1,23 @@
+Integration with fluent
+=======================
+
+Pontoon is able to synchronize translations produced by libraries provided by Project Fluent and provides
+advanced editor for translators.
+
+Because of our very close integration, we'll need to compile the fresh versions of javascript/python libraries
+in order to provide new features.
+
+It's important to remember to update both packages:
+* python-fluent (responsible for e.g. server-side sync process)
+* fluent-syntax (required by the fluent editor)
+
+How to build the fresh version of fluent-syntax.js
+--------------------------------------------------
+
+.. code-block:: bash
+    git clone https://github.com/projectfluent/fluent.js
+    cd fluent.js
+    npm install
+    cd fluent-syntax
+    make build
+    cp fluent-syntax.js <your pontoon directory>/pontoon/base/static/js/lib/fluent-syntax.js
```

Comment:
This might be an issue for another PR but I recommend against using custom builds of the master branch of `fluent-syntax`. We should stick to the official releases, either by using tags ([`fluent-syntax@0.5.0`](https://github.com/projectfluent/fluent.js/releases/tag/fluent-syntax%400.5.0)) or [npm](https://www.npmjs.com/package/fluent-syntax).  Also, it looks like Pontoon doesn't use any build system with transpilation at the moment, so I'd recommend using the compat builds of fluent which are transpiled with Babel for better compatibility.  As it is right now, the `fluent-syntax.js` that is included in production `pontoon.mozilla.org` uses a lot of es7 syntax and might break Pontoon in older browsers.

To sum up, I'd rephrase the above documentation to:

    cd <pontoon working dir>
    npm install fluent-syntax
    cp node_modules/fluent-syntax/compat.js pontoon/base/static/js/lib/fluent-syntax.js